src/components/calcite-date-picker/utils.ts

```diff
@@ -42,6 +42,9 @@ function getSupportedLocale(lang = "") {
 
   if (lang.includes("-")) {
     lang = lang.replace(/(\w+)-(\w+)/, (_match, language, region) => `${language}-${region.toUpperCase()}`);
+    if (!locales.includes(lang)) {
+      return locales.includes(lang.split("-")[0]) ? lang.split("-")[0] : "en";
```

Comment:
You could use the original return statement if you assign the language part to `lang`.

```ts
lang = lang.split("-")[0]; // the rest is handled by the existing return statement
```